extensions/admin-tool-ext-win/src/main.ts

```diff
@@ -7,82 +7,87 @@
 
 const baseConfig = require('./config.json');
 const localize = nls.loadMessageBundle();
-let exePath:string;
+let exePath: string;
+let processes: Map<number, ChildProcess> = new Map<number, ChildProcess>();
 
 // Params to pass to SsmsMin.exe, only an action and server are required - the rest are optional based on the
 // action used. Exported for use in testing.
 export interface LaunchSsmsDialogParams {
-    action: string;
-    server: string;
-    database?: string;
-    user?: string;
-    password?: string;
-    useAad?: boolean;
-    urn?: string;
+	action: string;
+	server: string;
+	database?: string;
+	user?: string;
+	password?: string;
+	useAad?: boolean;
+	urn?: string;
 }
 
 export async function activate(context: vscode.ExtensionContext): Promise<void> {
-    Telemetry.sendTelemetryEvent('startup/ExtensionActivated');
-
-    // Only supported on Win32 currently, display error message if not that until extensions are able to block install
-    // based on conditions
-    if(process.platform === 'win32') {
-        let config: IConfig = JSON.parse(JSON.stringify(baseConfig));
-        config.installDirectory = path.join(context.extensionPath, config.installDirectory);
-        config.proxy = utils.getConfiguration('http').get('proxy');
-        config.strictSSL = utils.getConfiguration('http').get('proxyStrictSSL') || true;
-
-        const serverdownloader = new ServerProvider(config);
-        const installationStart = Date.now();
-
-        try {
-            let downloadedExePath = await serverdownloader.getOrDownloadServer();
-            const installationComplete = Date.now();
-
-            // Don't register the command if we couldn't find the EXE since it won't be able to do anything
-            if(downloadedExePath) {
-                exePath = downloadedExePath;
-            } else {
-                throw new Error('Could not find SsmsMin.exe after downloading');
-            }
-            // Add the command now that we have the exePath to run the tool with
-            context.subscriptions.push(
-                vscode.commands.registerCommand('adminToolExtWin.launchSsmsServerPropertiesDialog', handleLaunchSsmsServerPropertiesDialogCommand));
-
-            Telemetry.sendTelemetryEvent('startup/ExtensionStarted', {
-                installationTime: String(installationComplete - installationStart),
-                beginningTimestamp: String(installationStart)
-            });
-        }
-        catch(err) {
-            Telemetry.sendTelemetryEvent('startup/ExtensionInitializationFailed', {
-                error: err
-            });
-        }
-    } else {
-        vscode.window.showErrorMessage(localize('adminToolExtWin.onlySupportedOnWindows', 'The Admin Tool Extension is only supported on Windows platforms.'));
-    }
+	// This is for Windows-specific support so do nothing on other platforms
+	if (process.platform === 'win32') {
+		Telemetry.sendTelemetryEvent('startup/ExtensionActivated');
```

Comment:
you may want to use a more specific event name.  It may make it easier to filter for this when reporting?

Reply:
All events have the extension name prefixed - so this will be admin-ext-win/startup/ExtensionActivated. That should be enough for our reporting purposes.